test/features/georoutingv2/georoutingv2.test.js

```diff
@@ -0,0 +1,470 @@
+        us: 'the United States',
+        de: 'Germany',
+        ch: 'Switzerland',
+        mena: 'the Middle East & North Africa',
+        africa: 'Africa',
+      },
+      {
+        prefix: 'africa',
+        us: 'the United States',
+        de: 'Germany',
+        ch: 'Switzerland',
+        mena: 'the Middle East & North Africa',
+        africa: 'Africa',
+      },
+    ],
+  },
+};
+
+let stubURLSearchParamsGet = stub(URLSearchParams.prototype, 'get');
+const setUserCountryFromIP = (country = 'CH') => {
+  stubURLSearchParamsGet = stubURLSearchParamsGet.withArgs('akamaiLocale').returns(country);
+};
+
+const ogFetch = window.fetch;
+window.fetch = stub();
+
+function stubHeadRequestToReturnVal(prefix, val) {
+  window.fetch.withArgs(`${prefix}`, { method: 'HEAD' }).returns(
+    new Promise((resolve) => {
+      resolve({ ok: val });
+    }),
+  );
+}
+
+const stubFetchForGeorouting = () => {
+  window.fetch.withArgs('/georoutingv2.json').returns(
+    new Promise((resolve) => {
+      resolve({
+        ok: true,
+        json: () => mockGeoroutingJson,
+      });
+    }),
+  );
+  mockGeoroutingJson.georouting.data.forEach((d) => {
+    const prefix = d.prefix ? `/${d.prefix}` : '';
+    stubHeadRequestToReturnVal(prefix, true);
+  });
+};
+const stubFallbackMetadata = (fallbackrouting) => {
+  getMetadata = stub();
+  getMetadata.withArgs('fallbackrouting').returns(fallbackrouting);
+};
+const restoreFetch = () => {
+  window.fetch = ogFetch;
+};
+const closeModal = () => {
+  document.querySelector('.dialog-modal')?.querySelector('.dialog-close').dispatchEvent(new Event('click'));
+};
+
+describe('GeoRouting', () => {
+  before(() => {
+    setUserCountryFromIP();
+    stubFetchForGeorouting();
+  });
+  after(() => {
+    stubURLSearchParamsGet.reset();
+    restoreFetch();
+  });
+  afterEach(() => {
+    document.cookie = 'international=; expires= Thu, 01 Jan 1970 00:00:00 GMT';
+    sessionStorage.removeItem('international');
+    closeModal();
+  });
+
+  it('Will read undefined if attempting to read unset cookie', async () => {
+    // prepare
+    const cookieName = 'test123';
+    const testcookie = getCookie(cookieName);
+    // assert
+    expect(testcookie).to.be.undefined;
+  });
+
+  it('Will read cookie value correctly if set', async () => {
+    // prepare
+    const cookieName = 'test';
+    document.cookie = `${cookieName}=test`;
+    const testcookie = getCookie(cookieName);
+    // assert
+    expect(testcookie).to.be.equal('test');
+    // cleanup
+    document.cookie = 'test=; expires= Thu, 01 Jan 1970 00:00:00 GMT';
+  });
+
+  it('Does create a modal if detected country from IP is CH and url prefix is US', async () => {
+    // prepare
+    await init(mockConfig, createTag, getMetadata, loadBlock, loadStyle);
+    const modal = document.querySelector('.dialog-modal');
+    // assert
+    expect(modal).to.not.be.null;
+  });
+
+  it('If session storage is us it is compared correctly', async () => {
```

Comment:
nit: `it` strings should start with verbs, and `us` here should be capitalized.

Maybe:
`Does not create a modal if the user's IP matches session storage`.

Reply:
ok, good to know - thx!